app/controllers/dashboard/giveaways_controller.rb

```diff
@@ -22,6 +22,8 @@ def index # rubocop:disable Metrics/AbcSize
     end
 
     def show
+      authorize @giveaway
```

Comment:
you can add this to a before_action filter specified for the actions used in so that you can clean up (dry) the code a bit

Reply:
Yes, I agree. That can be done in the controller I believe.